### PR TITLE
Road to v0.6.0

### PR DESCRIPTION
**Pull request by Globegitter**
I thought I'd get started on the work to `0.6.0`. These are the list of commits we need: https://github.com/graphql/graphql-js/compare/v0.5.0...v0.6.0

It seems some of those had already been done so I just added small bits as I thought necessary. Other than that I basically just added the suggestion of other graphql fields so far (That means all commit are done until and including #ec05b54)

Doesn't seem to be a huge milestone this time.

### CODE REVIEW

**Comment by Globegitter** [comment]
I have also started working on https://github.com/graphql/graphql-js/commit/e6e8d19cd7a8deb48a5a702500f9936e9fbc9c80 but it turns out that is not such an easy thing because of the way `schema._implementations` is a `defaultdict` rather than a normal `dict` here: https://github.com/graphql-python/graphql-core/blob/master/graphql/type/schema.py#L60

So it always returns a `list` where-as in graphql-js if a key does not exist it would return `undefined`. Imo the closes to that would be either using a `defaultdict` that returns `None` per default or just using a normal dict and then using the `.get` function on it.

Both of them break a lot of other unit tests, so not sure what the best approach forward is for this. Fixing all of the tests or just ignoring this commit.

**Comment by Globegitter** [comment]
Ok it actually wasn't too difficult to fix.Getting through the commits. Only 7 left.

**Comment by Globegitter** [comment]
@syrusakbary All commits are now moved over. For some reason my local `isort` does not give the same reulst as Travis' `isort` making this quite difficult for me. In any case, all other tests are passing.

**Comment by Globegitter** [comment]
Sweet, got all tests passing :)

**Comment by syrusakbary** [comment]
Hi Markus!

Thanks for your great work!
The upcoming features for this repo go in the `features/next` branch (some changes are required by the `next` graphene version, that's the reason is in a branch different than `master`).

If you could make another PR for the `features/next` branch with your improvements would be great! :)

**Comment by Globegitter** [comment]
@syrusakbary will do asap :)

**Comment by Globegitter** [comment]
@syrusakbary Merged in next branch, changed this PR (so did not have to open up a new one) and all tests are passing.

File: graphql/language/parser.py [hunk 1]
```diff
@@ -643,12 +662,14 @@ def parse_argument_defs(parser):
 def parse_input_value_def(parser):
     start = parser.token.start
 
-    return ast.InputValueDefinition(
+    bla = ast.InputValueDefinition(
```
Comment (syrusakbary) [inline graphql/language/parser.py:665]: bla? :P

**Comment by syrusakbary** [comment]
Looking good 👍

File: graphql/language/parser.py (hunk 1 above)
Comment (Globegitter) [reply graphql/language/parser.py:665]: Oh yes haha. Thank you for catching that @syrusakbary. That is one of my typical debugging variable names. Especially when I feel quite uncreative.

**Pull request merged**